src/mesh/index_derivs.cxx

```diff
@@ -1190,7 +1183,7 @@ const Field3D Mesh::indexDDX(const Field3D &f, CELL_LOC outloc, DIFF_METHOD meth
       throw BoutException("Cannot use FFT for X derivatives");
   }
 
-  result = applyXdiff(f, func, diffloc, region);
+  Field3D result = applyXdiff(f, func, diffloc, region);
 
   result.setLocation(diffloc); // Set the result location
```

Comment:
I know you've not touched this, but this looks suspicious. `applyXdiff` should be setting the location to `diffloc` -- do we need to then set the location again immediately after?

Reply:
I think this is actually hiding a bug at the moment, see last commit of #1061. If/when that is merged, we should probably get rid of the `result.setLocation` after `apply*diff`.